### PR TITLE
feat(make):  oauth flow, sync users 

### PR DESCRIPTION
**Pull request by awan7676**
**Title**
Integration of MAKE Tool

**Description**
This integration includes the implementation of authentication flow between the Elba and the MAKE and the corresponding tests.
It includes the implementation of the function required to fetch all the users of the MAKE tool and the complete test suite related to it.
It includes the implementation of Inngest functions for syncing users and scheduling users sync.
Implementation of rate-limit middleware with test suite is included.
All the tests related to specific functions are included.

### CODE REVIEW

**Comment by athielba** [comment]
@awan7676  This PR is only 3 weeks old but  it is not supporting the latest staging configuration, please update the PR to support the latest global support

File: apps/make/src/app/install/page.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
+'use client';
```
Comment (athielba) [inline apps/make/src/app/install/page.tsx:1]: UI seems not updated,  check [Datadog](https://github.com/elba-security/elba-security/blob/staging/apps/datadog/src/app/install/page.tsx)

File: apps/make/src/app/install/action.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    token: formData.get('token'),
+    organisationId: formData.get('organisationId'),
+    region: formData.get('region'),
+    zoneDomain: formData.get('zoneDomain'),
+  });
+  if (!result.success) {
+    const { fieldErrors } = result.error.flatten();
+    if (fieldErrors.organisationId || fieldErrors.region) {
+      return {
+        redirectUrl: `${env.ELBA_REDIRECT_URL}?source_id=${env.ELBA_SOURCE_ID}&error=internal_error`,
```
Comment (athielba) [inline apps/make/src/app/install/action.tsx:34]: Should use elba utils, check [Datadog ](https://github.com/elba-security/elba-security/blob/staging/apps/datadog/src/app/install/page.tsx)

File: apps/make/src/app/install/service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      },
+    });
+
+  await inngest.send({
+    name: 'make/users.sync.requested',
+    data: {
+      organisationId,
+      syncStartedAt: Date.now(),
+      isFirstSync: true
+    },
```
Comment (athielba) [inline apps/make/src/app/install/service.ts:38]: `make/app.installed` event is not called

File: apps/make/src/app/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,121 @@
+.container {
```
Comment (athielba) [inline apps/make/src/app/styles.module.css:1]: the css file is not necessary, since elba has UI utils

File: apps/make/src/database/client.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Pool, neonConfig } from '@neondatabase/serverless';
```
Comment (athielba) [inline apps/make/src/database/client.ts:1]:
```suggestion
import { Pool, neon, neonConfig } from '@neondatabase/serverless';
import type { NeonDatabase } from 'drizzle-orm/neon-serverless';
import { drizzle as drizzleNeonServerless } from 'drizzle-orm/neon-serverless';
import { drizzle as drizzleNeonHttp } from 'drizzle-orm/neon-http';
import { env } from '@/common/env';
import * as schema from './schema';

// eslint-disable-next-line import/no-mutable-exports -- to make it work locally
let db: NeonDatabase<typeof schema>;

if (!process.env.VERCEL_ENV || process.env.VERCEL_ENV === 'development') {
  neonConfig.wsProxy = (host) => `${host}:${env.DATABASE_PROXY_PORT}/v1`;
  neonConfig.useSecureWebSocket = false;
  neonConfig.pipelineTLS = false;
  neonConfig.pipelineConnect = false;

  const pool = new Pool({ connectionString: env.DATABASE_URL });
  db = drizzleNeonServerless(pool, { schema });
} else {
  // @ts-expect-error -- to make it work locally
  db = drizzleNeonHttp(neon(env.DATABASE_URL), { schema });
}

export { db };

```

File: apps/make/src/inngest/client.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+export const inngest = new Inngest({
+  id: 'make',
+  schemas: new EventSchemas().fromRecord<{
+    'make/users.sync.requested': {
+      data: {
+        organisationId: string;
+        syncStartedAt: number;
+      };
+    };
+    'make/users.page_sync.requested': {
```
Comment (athielba) [inline apps/make/src/inngest/client.ts:14]: should be `make/users.sync.requested`

File: apps/make/src/inngest/functions/users/schedule-users-syncs.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+export const scheduleUsersSyncs = inngest.createFunction(
+  { id: 'make-schedule-users-syncs' },
+  { cron: env.MAKE_USERS_SYNC_CRON },
+  async ({ step }) => {
+    const organisations = await db
+      .select({
+        id: Organisation.id,
+        token: Organisation.token,
+        zoneDomain: Organisation.zoneDomain,
+        region: Organisation.region,
```
Comment (athielba) [inline apps/make/src/inngest/functions/users/schedule-users-syncs.ts:15]: We don't need these
```javascript
        token: Organisation.token,
        zoneDomain: Organisation.zoneDomain,
        region: Organisation.region,

File: apps/make/src/inngest/functions/users/schedule-users-syncs.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    if (organisations.length > 0) {
+      await step.sendEvent(
+        'sync-organisations-users',
+        organisations.map(({ id }) => ({
+          name: 'make/users.sync.requested',
+          data: {
+            organisationId: id,
+            syncStartedAt: Date.now(),
+            isFirstSync: true
```
Comment (athielba) [inline apps/make/src/inngest/functions/users/schedule-users-syncs.ts:27]:
```suggestion
            isFirstSync: false
```

**Comment by awan7676** [comment]
Looking at these

File: apps/make/src/app/install/page.tsx (hunk 1 above)
Comment (awan7676) [reply apps/make/src/app/install/page.tsx:1]: Done

Comment (awan7676) [reply apps/make/src/app/install/page.tsx:1]: Done

File: apps/make/src/app/install/action.tsx (hunk 2 above)
Comment (awan7676) [reply apps/make/src/app/install/action.tsx]: Done

File: apps/make/src/app/styles.module.css (hunk 4 above)
Comment (awan7676) [reply apps/make/src/app/styles.module.css]: Deleted!

File: apps/make/src/database/client.ts (hunk 5 above)
Comment (awan7676) [reply apps/make/src/database/client.ts:1]: Done

File: apps/make/src/inngest/client.ts (hunk 6 above)
Comment (awan7676) [reply apps/make/src/inngest/client.ts:14]: Done

File: apps/make/src/inngest/functions/users/schedule-users-syncs.ts (hunk 7 above)
Comment (awan7676) [reply apps/make/src/inngest/functions/users/schedule-users-syncs.ts:15]: Done

File: apps/make/src/inngest/functions/users/schedule-users-syncs.ts (hunk 8 above)
Comment (awan7676) [reply apps/make/src/inngest/functions/users/schedule-users-syncs.ts:27]: Done

File: apps/make/src/connectors/make/users.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  if (page !== null) {
+    url.searchParams.append('pg[offset]', String(page));
+  }
+  const response = await fetch(url, {
+    headers: { Authorization: `Token ${token}` },
+  });
+  if (!response.ok) {
+    throw new MakeError('Could not retrieve users', { response });
+  }
+  const data = (await response.json()) as GetUsersResponseData;
```
Comment (athielba) [inline apps/make/src/connectors/make/users.ts:34]: Check datadog for getUser function, you have to  check the valid and invalid users

Comment (awan7676) [reply apps/make/src/connectors/make/users.ts]: Done

**Pull request merged**